pkg/provider/gke/gke.go

```diff
@@ -73,36 +73,35 @@ func (c *GKE) NewGKEClient(*kingpin.ParseContext) error {
 	// When the auth variable points to a file
 	// put the file content in the variable.
 	if content, err := ioutil.ReadFile(c.Auth); err == nil {
-		// Set the auth env variable needed to the k8s client.
-		// The client looks for this special variable name and it is the only way to set the auth for now.
-		// TODO Remove when the client supports an auth config option in NewDefaultClientConfig.
-		os.Setenv("GOOGLE_APPLICATION_CREDENTIALS", c.Auth)
 		c.Auth = string(content)
-	} else {
-		// Check if auth data is base64 encoded and decode it.
-		encoded, err := regexp.MatchString("^([A-Za-z0-9+/]{4})*([A-Za-z0-9+/]{3}=|[A-Za-z0-9+/]{2}==)?$", c.Auth)
+	}
+
+	// Check if auth data is base64 encoded and decode it.
+	encoded, err := regexp.MatchString("^([A-Za-z0-9+/]{4})*([A-Za-z0-9+/]{3}=|[A-Za-z0-9+/]{2}==)?$", c.Auth)
+	if err != nil {
+		return err
+	}
+	if encoded {
+		auth, err := base64.StdEncoding.DecodeString(c.Auth)
 		if err != nil {
 			return err
 		}
-		if encoded {
-			auth, err := base64.StdEncoding.DecodeString(c.Auth)
-			if err != nil {
-				return err
-			}
-			c.Auth = string(auth)
-		}
+		c.Auth = string(auth)
+	}
 
-		// Create tempory file to store the credentials.
-		saFile, err := ioutil.TempFile("", "service-account")
-		if err != nil {
-			log.Fatal(err)
-		}
-		defer saFile.Close()
-		if _, err := saFile.Write([]byte(c.Auth)); err != nil {
-			log.Fatal(err)
-		}
-		os.Setenv("GOOGLE_APPLICATION_CREDENTIALS", saFile.Name())
+	// Create tempory file to store the credentials.
+	saFile, err := ioutil.TempFile("", "service-account")
+	if err != nil {
+		log.Fatal(err)
+	}
+	defer saFile.Close()
+	if _, err := saFile.Write([]byte(c.Auth)); err != nil {
+		log.Fatal(err)
```

Comment:
Recently I learned that log.Fatal calles os.Exit and this way defer statments are never executed.
Maybe try to just return the error here.

I would say we should never use log.Fatal and refactor to return the error and handle it in the parent func.